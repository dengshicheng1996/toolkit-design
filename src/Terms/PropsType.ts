import { BasePropsType, BaseFormItemTypeWithOutFocus } from '../_internal'

export interface TermsPropsType
  extends BasePropsType,
    BaseFormItemTypeWithOutFocus<boolean> {
  // value for html form
  value?: string
  checked?: boolean
  defaultChecked?: boolean
  term: React.ReactNode

  // 在内容下侧
  describe?: boolean
  // 是否需要勾选
  hasChecked?: boolean
}
