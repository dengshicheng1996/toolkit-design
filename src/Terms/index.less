@import (reference) '../style/index.less';

// start
// 协议间距
@terms-spacing: @h-spacing-large;
// 普通协议
@terms-normal-size: 30 * @rpx;
@terms-normal-color: @color-text-title;
// 下至协议
@terms-desc-size: 22 * @rpx;
@terms-desc-color: @color-text-subtitle;

// 协议颜色
@terms-link-color: @color-brand-1;
// end

.amd-terms {
  padding: @terms-spacing;
  font-size: @font-size-content;
  line-height: @default-line-height;

  &-term {
    display: flex;
    margin: 0 0 40 * @rpx 0;
    color: @terms-normal-color;
    font-size: @terms-normal-size;

    .amd-checkbox {
      margin-right: 16 * @rpx;
    }

    a {
      color: @terms-link-color;
    }

    &-desc {
      margin: 14 * @rpx 0 0 0;
      color: @terms-desc-color;
      font-size: @terms-desc-size;
    }
  }
}
