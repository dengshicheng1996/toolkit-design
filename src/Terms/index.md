---
title: 协议
nav:
  title: 组件
  path: /components
---

### 展示

<code src="./demo/basic.tsx" />


### 参数

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| term | 协议内容 | React.ReactNode | - |
| describe | 是否描述性质 | boolean | false |
| hasChecked | 是否有 checkbox 表单元素 | boolean | false |
| checked | 受控 checked | boolean | - |
| defaultChecked | 非受控 defaultChecked | boolean | - |

> 当 hasChecked 打开，checked, defaultChecked 等表单元素才有效果, 其他属性见 BaseFormItemTypeWithoutFocus
