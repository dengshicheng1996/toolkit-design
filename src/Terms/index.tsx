import * as React from 'react'
import classnames from 'classnames'
import Checkbox from '../Checkbox'
import { withError } from '../rmc'
import { getDataAttr } from '../_internal'
import { useTracker } from '../hooks'
import { TermsPropsType } from './PropsType'

import './index.less'

const prefixCls = 'amd-terms'
export const Terms: React.FC<TermsPropsType> = ({
  className,
  term,
  hasChecked,
  describe,
  children,
  ...rest
}) => {
  useTracker(Terms.displayName)

  const node = (
    <div
      className={classnames(`${prefixCls}-term`, {
        [`${prefixCls}-term-desc`]: describe,
      })}
    >
      {hasChecked ? <Checkbox {...rest} /> : null}
      {term}
    </div>
  )

  return (
    <div {...getDataAttr(rest)} className={classnames(prefixCls, className)}>
      {describe ? null : node}
      <div className={`${prefixCls}-content`}>{children}</div>
      {describe ? node : null}
    </div>
  )
}

Terms.displayName = 'Terms'

Terms.defaultProps = {
  hasChecked: false,
  describe: false,
}

export default withError(Terms)
