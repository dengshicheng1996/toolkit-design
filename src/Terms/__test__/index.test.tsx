import * as React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { unstable_Button as Button } from '@ant-design/mobile'
import Terms from '..'

describe('Terms', () => {
  it('basic snapshot', () => {
    const component = mount(
      <Terms
        term={
          <>
            同意<a>《用户授权协议》</a>
          </>
        }
        data-y="111"
      >
        <Button type="primary">提交</Button>
      </Terms>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('checkbox snapshot', () => {
    const component = mount(
      <Terms
        hasChecked
        checked={false}
        term={
          <>
            同意<a>《用户授权协议》</a>
          </>
        }
      >
        <Button type="primary">提交</Button>
      </Terms>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('describe snapshot', () => {
    const component = mount(
      <Terms
        term={
          <>
            同意<a>《用户授权协议》</a>
          </>
        }
        describe
      >
        <Button type="primary">提交</Button>
      </Terms>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
