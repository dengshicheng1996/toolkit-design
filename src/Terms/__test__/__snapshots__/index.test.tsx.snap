// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terms basic snapshot 1`] = `
<Terms
  data-y="111"
  term={
    <React.Fragment>
      同意
      <a>
        《用户授权协议》
      </a>
    </React.Fragment>
  }
>
  <ErrorBoundary
    onError={[Function]}
  >
    <Terms
      data-y="111"
      describe={false}
      hasChecked={false}
      term={
        <React.Fragment>
          同意
          <a>
            《用户授权协议》
          </a>
        </React.Fragment>
      }
    >
      <div
        className="amd-terms"
        data-y="111"
      >
        <div
          className="amd-terms-term"
        >
          同意
          <a>
            《用户授权协议》
          </a>
        </div>
        <div
          className="amd-terms-content"
        >
          <Button
            type="primary"
          >
            <ErrorBoundary
              onError={[Function]}
            >
              <Button
                activeClassName=""
                capsule={false}
                capsuleAutoWidth={false}
                capsuleSize="md"
                className=""
                disabled={false}
                loading={false}
                subTitle=""
                type="primary"
              >
                <Touchable
                  disabled={false}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                >
                  <Touchable
                    disabled={false}
                    onPress={[Function]}
                    onPressIn={[Function]}
                    onPressOut={[Function]}
                  >
                    <View
                      onContextMenu={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <div
                        aria-disabled={false}
                        className="amd-button amd-button-primary"
                        onContextMenu={[Function]}
                        onTouchCancel={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="button"
                      >
                        <span
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          提交
                        </span>
                      </div>
                    </View>
                  </Touchable>
                </Touchable>
              </Button>
            </ErrorBoundary>
          </Button>
        </div>
      </div>
    </Terms>
  </ErrorBoundary>
</Terms>
`;

exports[`Terms checkbox snapshot 1`] = `
<Terms
  checked={false}
  hasChecked={true}
  term={
    <React.Fragment>
      同意
      <a>
        《用户授权协议》
      </a>
    </React.Fragment>
  }
>
  <ErrorBoundary
    onError={[Function]}
  >
    <Terms
      checked={false}
      describe={false}
      hasChecked={true}
      term={
        <React.Fragment>
          同意
          <a>
            《用户授权协议》
          </a>
        </React.Fragment>
      }
    >
      <div
        className="amd-terms"
      >
        <div
          className="amd-terms-term"
        >
          <Checkbox
            checked={false}
          >
            <ErrorBoundary
              onError={[Function]}
            >
              <Checkbox
                checked={false}
              >
                <Checkbox
                  checked={false}
                  defaultChecked={false}
                  prefixCls="amd-checkbox"
                  type="checkbox"
                >
                  <div
                    className="amd-checkbox"
                  >
                    <ForwardRef(C)
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    >
                      <ErrorBoundary
                        onError={[Function]}
                      >
                        <Labelable
                          checked={false}
                          forwardRef={null}
                          onChange={[Function]}
                          type="checkbox"
                        >
                          <label>
                            <input
                              checked={false}
                              onChange={[Function]}
                              readOnly={true}
                              style={
                                Object {
                                  "border": 0,
                                  "display": "block",
                                  "height": 0,
                                  "margin": 0,
                                  "overflow": "hidden",
                                  "padding": 0,
                                  "width": 0,
                                }
                              }
                              type="checkbox"
                            />
                            <div
                              className="amd-checkbox-fake"
                            />
                          </label>
                        </Labelable>
                      </ErrorBoundary>
                    </ForwardRef(C)>
                  </div>
                </Checkbox>
              </Checkbox>
            </ErrorBoundary>
          </Checkbox>
          同意
          <a>
            《用户授权协议》
          </a>
        </div>
        <div
          className="amd-terms-content"
        >
          <Button
            type="primary"
          >
            <ErrorBoundary
              onError={[Function]}
            >
              <Button
                activeClassName=""
                capsule={false}
                capsuleAutoWidth={false}
                capsuleSize="md"
                className=""
                disabled={false}
                loading={false}
                subTitle=""
                type="primary"
              >
                <Touchable
                  disabled={false}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                >
                  <Touchable
                    disabled={false}
                    onPress={[Function]}
                    onPressIn={[Function]}
                    onPressOut={[Function]}
                  >
                    <View
                      onContextMenu={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <div
                        aria-disabled={false}
                        className="amd-button amd-button-primary"
                        onContextMenu={[Function]}
                        onTouchCancel={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="button"
                      >
                        <span
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          提交
                        </span>
                      </div>
                    </View>
                  </Touchable>
                </Touchable>
              </Button>
            </ErrorBoundary>
          </Button>
        </div>
      </div>
    </Terms>
  </ErrorBoundary>
</Terms>
`;

exports[`Terms describe snapshot 1`] = `
<Terms
  describe={true}
  term={
    <React.Fragment>
      同意
      <a>
        《用户授权协议》
      </a>
    </React.Fragment>
  }
>
  <ErrorBoundary
    onError={[Function]}
  >
    <Terms
      describe={true}
      hasChecked={false}
      term={
        <React.Fragment>
          同意
          <a>
            《用户授权协议》
          </a>
        </React.Fragment>
      }
    >
      <div
        className="amd-terms"
      >
        <div
          className="amd-terms-content"
        >
          <Button
            type="primary"
          >
            <ErrorBoundary
              onError={[Function]}
            >
              <Button
                activeClassName=""
                capsule={false}
                capsuleAutoWidth={false}
                capsuleSize="md"
                className=""
                disabled={false}
                loading={false}
                subTitle=""
                type="primary"
              >
                <Touchable
                  disabled={false}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                >
                  <Touchable
                    disabled={false}
                    onPress={[Function]}
                    onPressIn={[Function]}
                    onPressOut={[Function]}
                  >
                    <View
                      onContextMenu={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <div
                        aria-disabled={false}
                        className="amd-button amd-button-primary"
                        onContextMenu={[Function]}
                        onTouchCancel={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="button"
                      >
                        <span
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          提交
                        </span>
                      </div>
                    </View>
                  </Touchable>
                </Touchable>
              </Button>
            </ErrorBoundary>
          </Button>
        </div>
        <div
          className="amd-terms-term amd-terms-term-desc"
        >
          同意
          <a>
            《用户授权协议》
          </a>
        </div>
      </div>
    </Terms>
  </ErrorBoundary>
</Terms>
`;
